### ISSUE TITLE
JSON encoding of SQS queue redrive policy includes extra whitespace

### ISSUE THREAD

**Issue by jessemyers** [localstack/localstack#3427]
# Type of request: This is a ...

[x ] bug report
[ ] feature request

# Detailed description

When using Terraform with Localstack to manage an SQS queue that contains a redrive policy, Localstack's JSON encoding of this redrive policy does not exactly match AWS's. As a result, the Terraform AWS provider currently detects a false diff.

IMO, this is a bug on both sides; the Terraform provider ought to parse the JSON as data, not as a string and the Localstack side should be more faithful to the AWS implementation to minimize compatibility issues.

You can find a bug report and steps to reproduce in this issue for the Terraform AWS provider:

https://github.com/hashicorp/terraform-provider-aws/issues/16964

## Expected behavior

The redrive policy is represented as JSON **without** whitespace between separators. For example:

```json
{"deadLetterTargetArn":"arn:aws:sqs:us-west-2:000000000000:example-DLQ","maxReceiveCount":3}
```

## Actual behavior

The json contains whitespaces after separators:

```json
{"deadLetterTargetArn": "arn:aws:sqs:us-west-2:000000000000:example-DLQ", "maxReceiveCount": 3}
```

# Steps to reproduce

```sh
docker-compose up
```

using this config

```yaml
version: '3.8'

volumes:
  localstack-data:

services:
  localstack:
    container_name: localstack_main
    image: localstack/localstack
    network_mode: bridge
    ports:
      - 4566:4566
    environment:
      - DEFAULT_REGION=us-west-2
      - SERVICES=sns,sqs
      - DATA_DIR=/tmp/localstack/data
    volumes:
      - localstack-data:/tmp/localstack
```

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)

I am using this `terraform` config:

```hcl
provider "aws" {
  access_key                  = ""
  region                      = "us-west-2"
  s3_force_path_style         = true
  secret_key                  = ""
  skip_credentials_validation = true
  skip_metadata_api_check     = true
  skip_requesting_account_id  = true

  endpoints {
    sqs = "http://localhost:4566"
  }
}

resource "aws_sqs_queue" "dead_letter_queue" {
  name                       = "example-DLQ"
  message_retention_seconds  = 14 * 24 * 60 * 60
  visibility_timeout_seconds = 1
}

resource "aws_sqs_queue" "queue" {
  name                       = "example"
  delay_seconds              = 0
  message_retention_seconds  = 14 * 24 * 60 * 60
  visibility_timeout_seconds = 1

  redrive_policy = jsonencode({
    deadLetterTargetArn = aws_sqs_queue.dead_letter_queue.arn
    maxReceiveCount     = 3
  })

  depends_on = [
    aws_sqs_queue.dead_letter_queue,
  ]
}
```

**Comment by adrian-hippo** [comment]
+1

**Comment by whummer** [comment]
Thanks for reporting @jessemyers . This should be fixed in #3451 - can you please pull the `latest` version and give it another try? Please report here if the problem persists. Thanks

**Comment by jessemyers** [comment]
Verified.

Thank you for your quick response.
